File: Plugin/src/main/java/dev/rosewood/roseskyblock/world/generator/ChunkLayer.java
package dev.rosewood.roseskyblock.world.generator;

import org.bukkit.Material;
import org.bukkit.generator.ChunkGenerator;

public record ChunkLayer(int startLayer, int endLayer, Material material) {

    public void fill(ChunkGenerator.ChunkData data) {
        data.setRegion(0, this.startLayer, 0, 16, this.endLayer + 1, 16, this.material);
    }

}

File: src/main/kotlin/dev/rosewood/roseskyblock/world/generator/ChunkLayer.kt
package dev.rosewood.roseskyblock.world.generator

import org.bukkit.Material
import org.bukkit.generator.ChunkGenerator

class ChunkLayer(val startLayer: Int, val endLayer: Int, val material: Material) {

    constructor(layer: Int, material: Material) : this(layer, layer, material)

    fun fill(chunkData: ChunkGenerator.ChunkData) = chunkData.setRegion(0, this.startLayer, 0, 16, this.endLayer + 1, 16, this.material)
}
